src/datepicker/datepicker.js

```diff
@@ -33,6 +35,13 @@ angular.module('ui.bootstrap.datepicker', ['ui.bootstrap.dateparser', 'ui.bootst
     self[key] = angular.isDefined($attrs[key]) ? (index < 6 ? $interpolate($attrs[key])($scope.$parent) : $scope.$parent.$eval($attrs[key])) : datepickerConfig[key];
   });
 
+  self.timelessJsonMode = angular.isDefined($attrs.timelessJsonMode) ?
+     $attrs.timelessJsonMode.toLowerCase() != 'false' && ($attrs.timelessJsonMode || true)
```

Comment:
It is opt-in. The `true` is there if the attribute is added without a value. It allows for `<div><input ng-model="date" uib-datepicker-popup="MM/dd/yyyy" timeless-json-mode></div>` to trigger the mode.